components/button-group/README.md

```diff
@@ -0,0 +1,51 @@
+# Button Groups
+
```

Comment:
I was considering this but since we handle elements like `d2l-link` I wasn't sure if it made sense. It does seem like link is the exception and in general its buttons/dropdown buttons that it will be handling.

Reply:
Yeah let's shelve this until after our meeting... I think we may land on something more generic than button-group anyway.